tabpy-server/server_tests/test_service_info_handler.py

```diff
@@ -120,3 +120,66 @@ def test_given_tabpy_server_with_auth_expect_correct_info_response(self):
                 'required': True,
             }
         }, features)
+
+
+class TestServiceInfoHandlerWithoutAuth(AsyncHTTPTestCase):
+    @classmethod
+    def setUpClass(cls):
+        prefix = '__TestServiceInfoHandlerWithoutAuth_'
+
+        # create state.ini dir and file
+        cls.state_dir = tempfile.mkdtemp(prefix=prefix)
+        cls.state_file = open(os.path.join(cls.state_dir, 'state.ini'), 'w+')
```

Comment:
I'd suggest you use a context manager for this, so:

```python
with open(os.path.join(cls.state_dir, 'state.ini'), 'w+') as state_file:
    state_file.write("...")
```

Implicitly closes for you and has better error handling

Reply:
fixed